tracing/tracing.go

```diff
@@ -25,22 +29,42 @@ func InitTracer(
 	endpoint string,
 	endpointType EndpointType,
 	samplingFraction float64,
-) (tp trace.TracerProvider, closer func(), err error) {
-	endpointOption := jaeger.WithAgentEndpoint(endpoint)
-	if endpointType == EndpointTypeCollector {
-		endpointOption = jaeger.WithCollectorEndpoint(endpoint)
-	}
+) (tp trace.TracerProvider, err error) {
+	tp = trace.NewNoopTracerProvider()
+
+	if endpoint != "" {
+		host, port, err := net.SplitHostPort(endpoint)
```

Comment:
Obviously this would return  the following error:
```
address http://localhost:14268/api/traces: too many colons in address
```
The problem here is that we provide a single endpoint for the agent as well as for the collector.

Reply:
I guess I meant to say whether we could check the `endpointType` beforehand and do `SplitHostPort` only for agent endpoint, as for collector endpoint we will hit this error.